### PR TITLE
docs: Update components.mdx

### PR DESCRIPTION
**Pull request by gilf**
Removed unused imports, added space lines for dividing the code and used prop destructuring instead of using a props object

# What is it?

- [ ] Feature / enhancement
- [ ] Bug
- [x] Docs / tests

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

# Use cases and why

- 1. One use case
- 2. Another use case

# Checklist:

- [x] My code follows the [developer guidelines of this project](../CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] Added new tests to cover the fix / functionality

### CODE REVIEW

File: packages/docs/src/routes/docs/cheat/components.mdx [hunk 1]
```diff
@@ -17,15 +17,16 @@ export const Greeter = component$(() => {
...
 interface GreeterProps {
   salutation?: string;
   name?: string;
 }
-export const Greeter = component$((props: GreeterProps) => {
-  const salutation = props.salutation || 'Hello';
-  const name = props.name || 'World';
+
+export const Greeter = component$(({ salutation, name }: GreeterProps) => {
+  const salutation = salutation || 'Hello';
```
Comment (shairez) [inline packages/docs/src/routes/docs/cheat/components.mdx:28]: @gilf wouldn't this type of assignment produce a `'salutation' has already been declared` error?

**Comment by shairez** [comment]
Thanks @gilf !

Added a comment

File: packages/docs/src/routes/docs/cheat/components.mdx (hunk 1 above)
Comment (gilf) [reply packages/docs/src/routes/docs/cheat/components.mdx:28]: Yep. Forgot to add default value in the destructing. Will change that and commit again.

Comment (shairez) [reply packages/docs/src/routes/docs/cheat/components.mdx]: Thanks!

**Pull request merged**